src/Entity/EmbedButton.php

```diff
@@ -128,7 +128,7 @@ public function getIconFile() {
    */
   public function getIconUrl() {
     if ($image = $this->getIconFile()) {
-      return $image->url();
```

Comment:
Would calling toUrl() and then casting returned instance of \Drupal\Core\Url to string work?